### PR TITLE
throw a useful error on invalid input to cheerio.load()

### PR DESCRIPTION
**Pull request by zeke**
👋 @jugglinmike 

This PR updates `cheerio.load()` to throw a more useful error message when bad input is given.

```
$ npm i -g trymodule && trymodule cheerio
Package 'cheerio' was loaded and assigned to 'cheerio' in the current scope
REPL started...

> cheerio.load(null)
TypeError: Cannot read property 'parent' of null
    at Function.exports.update (/Users/z/.trymodule/node_modules/cheerio/lib/parse.js:69:25)
    at module.exports (/Users/z/.trymodule/node_modules/cheerio/lib/parse.js:19:11)
    at Function.exports.load (/Users/z/.trymodule/node_modules/cheerio/lib/static.js:27:14)
    at repl:1:9
    at ContextifyScript.Script.runInContext (vm.js:53:29)
    at REPLServer.defaultEval (repl.js:241:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:120:20)
  ```

### CODE REVIEW

File: lib/static.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ var serialize = require('dom-serializer'),
  */
 
 exports.load = function(content, options) {
+  if (content === null || content === undefined) {
+    throw new TypeError('cheerio.load() expects a string');
```
Comment (fb55) [inline lib/static.js]: Please throw a plain error.

**Comment by zeke** [comment]
@fb55 I changed `TypeError` to `Error`. This is ready for review again.  👀

**Comment by fb55** [comment]
Awesome, thanks a lot! Will merge once tests pass.

File: test/api/traversing.js [hunk 2]
```diff
@@ -13,6 +13,17 @@ describe('$(...)', function() {
     $ = cheerio.load(fruits);
   });
 
+  describe('.load', function() {
+    it('should throw a TypeError if given invalid input', function() {
+      expect(function() {
+        cheerio.load();
+      }).to.throwException(function(err) {
+        expect(err).to.be.am(Error);
```
Comment (fb55) [inline test/api/traversing.js]: Typo: `an`

**Comment by zeke** [comment]
Thanks, @fb55!

**Pull request merged**